### PR TITLE
Update custom Doclet configuration for javadoc

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Starting with version 3.0 `javadoc` uses [updated parameter names](https://maven.apache.org/plugins/maven-javadoc-plugin/examples/alternate-doclet.html) in custom Doclet configurations, which affects our internal devsite scripts.
-  This PR updates the config to use the new parameters names.
- Tested and verified with the devsite script.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -123,16 +123,16 @@
...
-                                -hdf book.path /docs/reference/_book.yaml
-                                -hdf project.path /_project.yaml
-                                -hdf devsite.path /docs/reference/admin/java/reference/
-                                -d ${project.build.directory}/apidocs
-                                -templatedir ${devsite.template}
-                                -toroot /docs/reference/admin/java/reference/
-                                -yaml _toc.yaml
-                                -warning 101
-                            </additionalparam>
+                            <additionalOptions>
```
Comment (hiranya911) [inline pom.xml:126]: I think there are couple of other places in this file that use the old syntax. Can we change them too?

**Review by hiranya911 — CHANGES_REQUESTED** [review]
Thanks for putting this together. Just one suggestion.

File: pom.xml (hunk 1 above)
Comment (lahirumaramba) [reply pom.xml:126]: Wasn't sure if the rest should be updated as well. Thanks! I will update all the instances.

**Comment by lahirumaramba** [comment]
Looks like line 87 still uses the current config name. Updated the rest. Thanks!

**Comment by hiranya911** [comment]
Thanks. May be we can remove that profile entirely, and just merge the configuration with the rest? It's reasonable to require jdk 1.8 or higher for all build and release tasks:

```xml
       <profile>
            <!-- Disable Doclint on Java 8 and higher -->
            <id>disable-java8-doclint</id>
            <activation>
                <jdk>[1.8,)</jdk>
            </activation>
            <properties>
                <additionalparam>-Xdoclint:none</additionalparam>
            </properties>
        </profile>
```

File: pom.xml [hunk 2]
```diff
@@ -123,16 +113,16 @@
...
+                                <additionalOption>-hdf book.path /docs/reference/_book.yaml</additionalOption>
```
Comment (hiranya911) [inline pom.xml:117]: We need to pass the `-Xdoclint:none` option here too. Earlier it was provided by the profile whenever JDK 1.8 was available. Now we have to explicitly set it.

**Review by hiranya911 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (lahirumaramba) [reply pom.xml:117]: Thanks! Added `<doclint>none</doclint>`

File: pom.xml [hunk 3]
```diff
@@ -334,9 +325,9 @@
                             <version>1.3</version>
                         </additionalDependency>
                     </additionalDependencies>
-                    <additionalparam>
-                        -warning 101
-                    </additionalparam>
+                    <additionalOptions>
+                        <additionalOption>-warning 101</additionalOption>
+                    </additionalOptions>
```
Comment (hiranya911) [inline pom.xml:330]: Probably need doclint:none here too.

**Review by hiranya911 — APPROVED** [review]
LGTM with a suggestion.

File: pom.xml (hunk 3 above)
Comment (lahirumaramba) [reply pom.xml:330]: Updated. Thanks!

**Pull request merged**